### PR TITLE
Buffered mjpeg file writer

### PR DESCRIPTION
**Pull request by tonihele**
This optimizes the VideoRecorderAppState video file writing.

- The old version writes really small chunks, 4 bytes, even 1 byte chunks at times, PER FRAME!
- The new version uses buffer (8192 bytes) to write bigger chunks
- In reality much of the data written at a time (the individual frames) is a bigger chunk than 8192 -> written at one go
- Implements AutoCloseable, in my mind this is logical. Like writing a ZIP file etc. similar API

Basically this improves performance by eliminating some of the system calls.

New:
image

Old:
image

### CODE REVIEW

**Comment by stephengold** [comment]
Thanks for your contribution.
